gpuallocator/device.go

```diff
@@ -7,61 +7,96 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml"
+	"github.com/NVIDIA/go-gpuallocator/internal/links"
+	"gitlab.com/nvidia/cloud-native/go-nvlib/pkg/nvlib/device"
+	"gitlab.com/nvidia/cloud-native/go-nvlib/pkg/nvml"
 )
 
 // Device represents a GPU device as reported by NVML, including all of its
 // Point-to-Point link information.
 type Device struct {
-	*nvml.Device
+	device.Device
 	Index int
 	Links map[int][]P2PLink
+	// We cache certain values for the device.
```

Comment:
I would like to avoid embedding all these public symbols, but wanted to limit changes to the other code.